src/pages/workspaces/workspace/WorkspaceMenu.test.ts

```diff
@@ -139,12 +171,24 @@ describe('WorkspaceMenu - defined workspace (GCP or Azure)', () => {
     { menuText: 'Unlock', accessLevel: 'READER' },
   ])('enables/disables $menuText menu item based on access level $accessLevel', ({ menuText, accessLevel }) => {
     // Arrange
-    useWorkspaceDetails.mockReturnValue({ workspace: { workspace: { isLocked: menuText === 'Unlock' }, accessLevel } });
+    asMockedFn(useWorkspaceDetails).mockReturnValue({
+      // @ts-expect-error - the type checker thinks workspace is only of type undefined
+      workspace: {
+        ...defaultGoogleWorkspace,
+        accessLevel: accessLevel as WorkspaceAccessLevel,
+        workspace: {
+          ...defaultGoogleWorkspace.workspace,
+          isLocked: menuText === 'Unlock',
+        },
+      },
+      refresh: jest.fn(),
+      loading: false,
+    });
     // Act
     render(h(WorkspaceMenu, workspaceMenuProps));
     const menuItem = screen.getByText(menuText);
     // Assert
-    if (WorkspaceUtils.isOwner(accessLevel)) {
+    if (WorkspaceUtils.isOwner(accessLevel as WorkspaceAccessLevel)) {
```

Comment:
Suggest casting the array of arguments to `it.each` to avoid having to cast in the test. I _think_ that will also check that the provided access levels are valid.

```ts
it.each([
    { menuText: 'Lock', accessLevel: 'OWNER' },
    { menuText: 'Lock', accessLevel: 'READER' },
    { menuText: 'Unlock', accessLevel: 'OWNER' },
    { menuText: 'Unlock', accessLevel: 'READER' },
  ] as { menuText: string; accessLevel: WorkspaceAccessLevel }[])(...
```